lib/manager/terraform/extract.ts

```diff
@@ -58,128 +33,59 @@ export function extractPackageFile(content: string): PackageFile | null {
-        dep.managerData.source
-      );
-      /* eslint-disable no-param-reassign */
-      if (githubRefMatch) {
-        const depNameShort = githubRefMatch[2].replace(/\.git$/, '');
-        dep.depType = 'github';
-        dep.depName = 'github.com/' + depNameShort;
-        dep.depNameShort = depNameShort;
-        dep.currentValue = githubRefMatch[3];
-        dep.datasource = datasourceGithubTags.id;
-        dep.lookupName = depNameShort;
-        if (!isVersion(dep.currentValue)) {
-          dep.skipReason = SkipReason.UnsupportedVersion;
-        }
-      } else if (gitTagsRefMatch) {
-        dep.depType = 'gitTags';
-        if (gitTagsRefMatch[2].includes('//')) {
-          logger.debug('Terraform module contains subdirectory');
-          dep.depName = gitTagsRefMatch[2].split('//')[0];
-          dep.depNameShort = dep.depName.split(/\/(.+)/)[1];
-          const tempLookupName = gitTagsRefMatch[1].split('//');
-          dep.lookupName = tempLookupName[0] + '//' + tempLookupName[1];
-        } else {
-          dep.depName = gitTagsRefMatch[2].replace('.git', '');
-          dep.depNameShort = gitTagsRefMatch[3].replace('.git', '');
-          dep.lookupName = gitTagsRefMatch[1];
-        }
-        dep.currentValue = gitTagsRefMatch[4];
-        dep.datasource = datasourceGitTags.id;
-        if (!isVersion(dep.currentValue)) {
-          dep.skipReason = SkipReason.UnsupportedVersion;
-        }
-      } else if (dep.managerData.source) {
-        const moduleParts = dep.managerData.source.split('//')[0].split('/');
-        if (moduleParts[0] === '..') {
-          dep.skipReason = SkipReason.Local;
-        } else if (moduleParts.length >= 3) {
-          dep.depType = 'terraform';
-          dep.depName = moduleParts.join('/');
-          dep.depNameShort = dep.depName;
-          dep.datasource = datasourceTerraformModule.id;
-        }
-      } else {
-        logger.debug({ dep }, 'terraform dep has no source');
-        dep.skipReason = SkipReason.NoSource;
-      }
-    } else if (
-      dep.managerData.terraformDependencyType ===
-        TerraformDependencyTypes.provider ||
-      dep.managerData.terraformDependencyType ===
-        TerraformDependencyTypes.required_providers
-    ) {
-      dep.depType = 'terraform';
-      dep.depName = dep.managerData.moduleName;
-      dep.depNameShort = dep.managerData.moduleName;
-      dep.datasource = datasourceTerraformProvider.id;
-      if (!isValid(dep.currentValue)) {
-        dep.skipReason = SkipReason.UnsupportedVersion;
-      }
+    switch (dep.managerData.terraformDependencyType) {
+      case TerraformDependencyTypes.required_providers:
+      case TerraformDependencyTypes.provider:
+        analyzeTerraformProvider(dep);
+        break;
+      case TerraformDependencyTypes.module:
+        analyseTerraformModule(dep);
+        break;
+      /* istanbul ignore next */
+      default:
```

Comment:
Do we really need the default?

Reply:
This is a requirement of ES-lint, without it the linting fails.